### ISSUE TITLE
Polymer 2 - modulesLoaded.some is not a function

### ISSUE THREAD

**Issue by ahrobinson** [PredixDev/px-view#11]
on line 239, modulesLoaded.some is called. However, on inspection, modulesLoaded is actually a NodeList and not an array, and does not have access to the .some function.

This can be resolved by doing Array.prototype.some.call(modulesLoaded, function(){})

Could I make a PR for this?

**Comment by gstroup** [comment]
Hmm... I haven't noticed this problem.  How can I reproduce it?  Your solution makes sense, I'm just surprised no one else has seen this problem.  Thanks.

**Comment by ahrobinson** [comment]
We were using Polymer 2 at the time (and decided to go back to Polymer 1) and it is now working. So I'm guessing that is the reason, as Polymer 2 is supposed to be straight to web components spec.

Upon research document.querySelectorAll returns a NodeList [(MDN)](https://developer.mozilla.org/en-US/docs/Web/API/NodeList), but in Polymer 1 they purposely return an Array [(Polymer 1 docs)](https://www.polymer-project.org/1.0/docs/devguide/local-dom#dom-api) 
> Properties and methods that return a list of nodes return an Array, not a NodeList.

I can, however still try and reproduce when I have more time.

**Comment by gstroup** [comment]
Ahh, that explains it.  Thanks for researching!  I know that some of the px components will need to be updated to support Polymer 2.  Looks like this is one of them.  No need for you to spend any more time on it.  I'll leave this issue open as a reminder.  Thanks.

**Comment by cherukumilli** [comment]
@gstroup 
We started migrating one of our micro app's from Polymer 1.x to Polymer 2.x.
I see the error stated above by `ahrobinson`.

Looks like `ahrobinson` is right. 

## on Line 239,
In polymer 1.x, `Polymer.dom(...)` is returning an `Array`
In polymer 2.x, `Polymer.dom(...)` is returning a `NodeList`

I tested @ahrobinson 's fix in both polymer 1.x and 2.x and it is `working`

basically as @ahrobinson suggested we just need to replace the existing line 239 with the following:
```
if (Array.prototype.some.call(modulesLoaded, function(ele) {return ele.id === that._elementTagName;})) {
```

Please let me know if you would entertain a PR for this fix.

**Comment by gstroup** [comment]
Yes, we could accept a PR for this fix.  Thanks for digging in to the details.

**Comment by cherukumilli** [comment]
@gstroup - I don't think I have permissions to contribute to this repo. Can you pls make me a contributor?

**Comment by gstroup** [comment]
I think I've given you the right permissions now.  Can you make a PR from your fork?   Thanks for the help!
